### PR TITLE
improve acl handler

### PR DESCRIPTION
**Pull request by adameat**
### Changelog entry 

Improved ACL handler:
* Instead of SchemeShard we make calls to SchemeCache which is faster and more reliable
* In addition to ACL we also return EffectiveACL list
* New parameter merge_rules allows backend to merge rights into rules to make response more easy to read

### Changelog category 

* Improvement
* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/core/viewer/protos/viewer.proto [hunk 1]
```diff
@@ -91,6 +92,7 @@ message TMetaCommonInfo {
     repeated uint32 Nodes = 15;
     repeated uint32 Disks = 16;
     repeated TACE ACL = 17;
+
```
Comment (StekPerepolnen) [inline ydb/core/viewer/protos/viewer.proto:95]: unnecessary line

File: ydb/core/viewer/json_acl.h [hunk 2]
```diff
@@ -32,206 +33,271 @@ class TJsonACL : public TViewerPipeClient<TJsonACL> {
...
+                                  type: array
+                                  items:
+                                    type: string
+            400:
+              description: Bad Request
+            403:
+              description: Forbidden
+            504:
+              description: Gateway Timeout
+
```
Comment (StekPerepolnen) [inline ydb/core/viewer/json_acl.h:296]: it could return 307 Temporary Redirect, 500 Internal Error also

File: ydb/core/viewer/json_acl.h [hunk 3]
```diff
@@ -32,206 +33,271 @@ class TJsonACL : public TViewerPipeClient<TJsonACL> {
...
-                        }
-                    }
+    static bool Has(ui32 accessRights, ui32 mask) {
+        return (accessRights & mask) == mask;
+    }
+
+    void FillACE(const NACLibProto::TACE& ace, NKikimrViewer::TMetaCommonInfo::TACE& pbAce) {
+        if (static_cast<NACLib::EAccessType>(ace.GetAccessType()) == NACLib::EAccessType::Deny) {
+            pbAce.SetAccessType("Deny");
+        } else
```
Comment (StekPerepolnen) [inline ydb/core/viewer/json_acl.h:83]: we put "else if" in one line

File: ydb/core/viewer/json_acl.h [hunk 4]
```diff
@@ -32,206 +33,271 @@ class TJsonACL : public TViewerPipeClient<TJsonACL> {
...
+            {NACLib::EAccessRights::ConnectDatabase, "ConnectDatabase"},
+            {NACLib::EAccessRights::ReadStream, "ReadStream"},
+            {NACLib::EAccessRights::WriteStream, "WriteStream"},
+            {NACLib::EAccessRights::ReadTopic, "ReadTopic"},
+            {NACLib::EAccessRights::WriteTopic, "WriteTopic"}
+        };
+        for (const auto& [right, name] : accessRights) {
+            if (Has(ar, right)) {
+                pbAce.AddAccessRights(name);
+                ar &= ~right;
```
Comment (StekPerepolnen) [inline ydb/core/viewer/json_acl.h:134]: It's not necessary line

File: ydb/core/viewer/protos/viewer.proto (hunk 1 above)
Comment (adameat) [reply ydb/core/viewer/protos/viewer.proto:95]: it makes it easier to read

File: ydb/core/viewer/json_acl.h (hunk 2 above)
Comment (adameat) [reply ydb/core/viewer/json_acl.h:296]: there is no requirement to describe all of them

**Review by StekPerepolnen — APPROVED** [review]

**Pull request merged**